ui/src/BsddTreeView.js

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import TreeView from '@mui/lab/TreeView';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import TreeItem from '@mui/lab/TreeItem';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import { statusToColor } from './mappings'
+
+export default function BsddTreeView({ bsddResults, status }) {
+
+  const bsdd = bsddResults;
+
+  return (
+    <Paper sx={{overflow: 'hidden'}}><TreeView
+      aria-label="file system navigator"
+      defaultCollapseIcon={<ExpandMoreIcon />}
+      defaultExpandIcon={<ChevronRightIcon />}
+      defaultExpanded={["0"]}
+      sx={{ "width": "850px", "backgroundColor": statusToColor[status], "> li > .MuiTreeItem-content": { padding: "16px" }, ".MuiTreeItem-content.Mui-expanded": { borderBottom: 'solid 1px black' } }}
+    >
+      <TreeItem nodeId="0" label={"bSDD"}>
+        <TreeView defaultCollapseIcon={<ExpandMoreIcon />}
+          defaultExpandIcon={<ChevronRightIcon />}>
+          {
+            Object.keys(bsdd["bsdd"]).map((domain, d_index) => {
+              return <TreeItem nodeId={d_index} label={`Domain: ${domain}`}>
+                <TreeView defaultCollapseIcon={<ExpandMoreIcon />}
+                  defaultExpandIcon={<ChevronRightIcon />}>
+                  {
+                    Object.keys(bsdd["bsdd"][domain]).map((classification, c_index) => {
```

Comment:
Instead of doing `Object.keys(bsdd["bsdd"][domain]).map()` and then in the function body again having to access `bsdd["bsdd"][domain]` because you only have the key.

Why not do Object.entries().map([k,v] => {}) so that you iterate over the pairs and get the corresponding value immediately.

Also, I really prefer the dot notation in javascript where possible. `bsdd["bsdd"]` -> bsdd.bsdd

Reply:
Beautiful